### PR TITLE
bin: custom git commands

### PR DESCRIPTION
**Pull request by joyeecheung**
This was https://github.com/nodejs/node-core-utils/pull/98 , opening a new PR because the repo has been transferred so I need to update the branch of my fork instead.

### CODE REVIEW

File: bin/git/git-node-apply [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env node
+'use strict';
+
+const PRID = parseInt(process.argv[2]);
+const Request = require('./../../lib/request');
+const paths = require('./../../lib/paths');
```
Comment (cPhost) [inline bin/git/git-node-apply]: I think for both `Request` and `paths` you can remove `./`
nit: `../../lib/request` and `../../lib/paths`

File: bin/git/git-node-apply [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const { run, runAsync } = require('../../lib/run');
+const fs = require('fs');
+const CLI = require('../../lib/CLI');
```
Comment (cPhost) [inline bin/git/git-node-apply]: I think this should be `../../lib/cli`.

**Comment by joyeecheung** [comment]
Note that this is still WIP, I would still need to find some time to port the rest of the bash scripts to Node.js scripts. Although I have been using this fork locally to land stuff anyways.

**Comment by cPhost** [comment]
I totally did not mean to hit approve meant to comment...

**Comment by cPhost** [comment]
@joyeecheung tried to port bash scripts to node-script and this is currently what i have.
https://gist.github.com/cPhost/dec7ce68a09907791ce99147feb618e9

(I this help great! 👍 )

**Comment by joyeecheung** [comment]
@cPhost Thanks, but the porting is not really what's blocking me right now...The more I use it, the more I prefer the `git node land --continue` concept, so the commands kinda need a rework.

**Comment by cPhost** [comment]
Also after landing this before publishing to npm, we need to ensure the Windows compatibility.

**Comment by cPhost** [comment]
After thinking about sub commands i think we should have then and and implement single land script that does all the things at once with the subcommands.

**Comment by joyeecheung** [comment]
Recording of working prototype: https://asciinema.org/a/6i3gFDSAr7L6fYPSEaMrW0asO

**Comment by joyeecheung** [comment]
I've been using it for about two months now and it seems to be working pretty well. There are still a few things that need to be addressed:

1. `core-validate-commit` must be installed
2. `core-validate-commit` still complains about commit titles > 50 char so `git node land --final` fails from time to time. (should be fixed by https://github.com/nodejs/core-validate-commit/pull/18)
3. Needs some docs

I think when I finished 3 this should be ready to land on master.

**Comment by joyeecheung** [comment]
Docs done. Demos:

Landing multiple commits: https://asciinema.org/a/148627
Landing one commit: https://asciinema.org/a/157445

This should be ready to merge.

**Comment by cPhost** [comment]
This Looks good to me :)

**Comment by joyeecheung** [comment]
I am going to merge this and release a new version. Let's see how it works in the wild.

**Pull request merged**